src/lib/support/ThreadOperationalDataset.h

```diff
@@ -47,6 +47,8 @@ inline constexpr size_t kSizePSKc            = 16;
 class OperationalDataset
 {
 public:
+    OperationalDataset() : mLength(0) /* mData does not need initialization */ {}
```

Comment:
What you did here has no difference than what was there before. Why is this change done like this?

Reply:
```
INFO    ../../examples/all-clusters-app/tizen/third_party/connectedhomeip/src/app/clusters/thread-network-directory-server/thread-network-directory-server.cpp: In instantiation of ‘chip::app::Clusters::ThreadNetworkDirectoryServer::ReadThreadNetworks(const chip::app::ConcreteDataAttributePath&, chip::app::AttributeValueEncoder&)::<lambda(const auto:9&)> [with auto:9 = chip::app::AttributeValueEncoder::ListEncodeHelper; CHIP_ERROR = chip::ChipError]’:
INFO    ../../examples/all-clusters-app/tizen/third_party/connectedhomeip/src/app/AttributeValueEncoder.h:133:35:   required from ‘CHIP_ERROR chip::app::AttributeValueEncoder::EncodeList(ListGenerator) [with ListGenerator = chip::app::Clusters::ThreadNetworkDirectoryServer::ReadThreadNetworks(const chip::app::ConcreteDataAttributePath&, chip::app::AttributeValueEncoder&)::<lambda(const auto:9&)>; CHIP_ERROR = chip::ChipError]’
INFO    ../../examples/all-clusters-app/tizen/third_party/connectedhomeip/src/app/clusters/thread-network-directory-server/thread-network-directory-server.cpp:158:6:   required from here
INFO    ../../examples/all-clusters-app/tizen/third_party/connectedhomeip/src/lib/support/ThreadOperationalDataset.h:47:7: note: ‘class chip::Thread::OperationalDataset’ has no user-provided default constructor
INFO       47 | class OperationalDataset
INFO          |       ^~~~~~~~~~~~~~~~~~
INFO    ../../examples/all-clusters-app/tizen/third_party/connectedhomeip/src/lib/support/ThreadOperationalDataset.h:336:13: note: and the implicitly-defined constructor does not initialize ‘uint8_t chip::Thread::OperationalDataset::mData [254]’
INFO      336 |     uint8_t mData[kSizeOperationalDataset];
INFO          |             ^~~~~
```

^ This seems to be treated as an error on some toolchains with `-Werror`

Reply:
I've taken it out for now as this issue does not appear on the Linux builds.